client/src/components/HomePage.js

```diff
@@ -0,0 +1,78 @@
+import React, { useState, useEffect } from "react"
+import RestaurantTile from "./RestaurantTile"
+import { Link } from "react-router-dom"
+
+const HomePage = (props) => {
+    const [restaurants, setRestaurants] = useState([])
+
+    const getRestaurants = async () => {
+        try {
+            const response = await fetch("/api/v1/restaurants")
+            if (!response.ok) {
+                throw new Error(`${response.status} (${response.statusText})`)
+            }
+            const body = await response.json()
+            setRestaurants(body.restaurants)
+        } catch (error) {
+            console.error(`Error in fetch: ${error.message}`)
+        }
+    }
+
+    useEffect(() => {
+        getRestaurants()
+    }, [])
+
+    const top3Restaurants = () => {
+        let allRestaurants = restaurants
+        allRestaurants.sort((a, b) => {
+            return b.averageRating - a.averageRating
+        })
+
+        let top3 = []
+        if (allRestaurants.length > 3) {
+            for (let i = 0; i < 3; i++) {
+                top3.push(allRestaurants[i])
+            } 
+        } else {
+            top3 = allRestaurants
+        }
+
+        return top3
+    }
```

Comment:
data manipulation should be handled on the backend - there is no need for the fetch request to return all the results in the database (potentially thousands of records), which is computationally expensive. It also then take a lot for React to sort through and find the ones it wants.

Any type of data manipulation like this should be done in the router so that the back end only sends data to the front end that is going to be used